File: 230-kth-smallest-element-in-a-bst/230-kth-smallest-element-in-a-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        collection = []

        def extract(node):
            if not node:
                return

            extract(node.left)
            collection.append(node.val)
            extract(node.right)

        extract(root)
        return collection[k-1]

File: 230-kth-smallest-element-in-a-bst/230-kth-smallest-element-in-a-bst.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution
{
public:
  vector<int> res;
  int kthSmallest(TreeNode *root, int k)
  {
    helper(root);
    return res[k - 1];
  }
  void helper(TreeNode *root)
  {
    if (root == nullptr)
    {
      return;
    }
    if (root->left)
    {

      helper(root->left);
    }
    res.push_back(root->val);
    if (root->right)
    {

      helper(root->right);
    }
  }
};